### ISSUE TITLE
Test containers taking very long to start

### ISSUE THREAD

**Issue by shumonsharif** [quarkiverse/quarkus-cxf#523]
Local builds of the project on my MacBook Pro are very inconsistent - approx. 80% of the time, builds fail due to the test containers not starting properly within the allotted time of 60 seconds.

**Comment by ppalaga** [comment]
Ups, sad to hear that. I admit the containers are not super fast to boot.
They are based on WildFly Quickstarts, which was just the way I knew best. 
If you see a way to make them boot faster, e.g. by using another container, your suggestions/PRs are welcome. 

What is the specs of you machine, BTW?

**Comment by shumonsharif** [comment]
Hey @ppalaga I think I narrowed down the issue.  I am on an M1 chip for my MacBook, and it seems to be related to the below, quoting from the [docker website](https://docs.docker.com/desktop/mac/apple-silicon/):

> Some images do not support the ARM64 architecture. You can add --platform linux/amd64 to run (or build) an Intel image using emulation.

> However, attempts to run Intel-based containers on Apple silicon machines under emulation can crash as qemu sometimes fails to run the container. In addition, filesystem change notification APIs (inotify) do not work under qemu emulation. Even when the containers do run correctly under emulation, they will be slower and use more memory than the native equivalent.

> In summary, running Intel-based containers on Arm-based machines should be regarded as “best effort” only. We recommend running arm64 containers on Apple silicon machines whenever possible, and encouraging container authors to produce arm64, or multi-arch, versions of their containers. We expect this issue to become less common over time, as more and more images are rebuilt [supporting multiple architectures](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/).

I see the [latest wildfly image on quay](https://quay.io/repository/wildfly/wildfly?tab=tags) has multi-arch / arm64 support ... though not yet sure how that can be integrated in our test flows.

**Comment by ppalaga** [comment]
We use buildah to build the calculator-ws image on GitHub actions: https://github.com/l2x6/calculator-ws/blob/1.0/.github/workflows/deploy.yaml#L19-L26 

It looks like buildah can build for arm64 as well: https://medium.com/oracledevs/building-multi-architecture-containers-with-buildah-44ed100ec3f3 

If you'd like to make it work a contribution would be welcome.

**Comment by shumonsharif** [comment]
Let me try take a look at this!

**Comment by shumonsharif** [comment]
Just submitted https://github.com/l2x6/calculator-ws/pull/5 - this should hopefully address the issue.  

Also, there should certainly be some ways to speed this up, such as not depending on a full wildfly distro; though I think this should be good enough for now!

**Comment by ppalaga** [comment]
l2x6/calculator-ws#5 is merged. You may want to `docker pull quay.io/l2x6/calculator-ws:1.0` and check whether it works better for you

**Comment by ppalaga** [comment]
>  No need for a separate docker pull as it automatically knows which image to pull

Yeah, but the image might have been cached locally. If you pull you can be sure that it is the one from quay rather than e.g. the one you built locally.

**Comment by shumonsharif** [comment]
Yes, I did clear out all the local images prior to running `mvn clean package` ... I was making sure that the tests would execute successfully for any contributors who may try to build the project from a fresh clone, and more importantly, that the correct version of the multi-arch image is pulled.
